### ISSUE TITLE
Implement a function to add ERC-20 tokens

### ISSUE THREAD

**Issue by Dprof-in-tech** [dewdrop-labs/transverse-contracts#11]
Implement a function mechanism to add supported ERC-20 tokens into the contract.

**Comment by olisaagbafor** [comment]
Can I take this from here?

**Comment by Levi-Ojukwu** [comment]
Hello Dprof-in-tech, I will like to work on this issue.

## About Me 
I am a web3 full-stack developer with expertise in blockchain technology and frontend engineering. I build on EVM-compatible (Solidity) and Starknet (Cairo) blockchains.
In the blockchain domain, my expertise in Solidity and Cairo has enabled me to design, develop, and deploy smart contracts on Ethereum and Starknet-based platforms, facilitating decentralized applications (dApps) that are revolutionizing industries. 

## How to tackle the issue
- To tackle this issue, I will start by creating a function named `addERC20Token`
- Then I will proceed by running a check if the contract supports ERC20 tokens 
- Then I will implement a function to add ERC20 token to the function.
- I will then write a test for the funcion, testing and ensuring all scenarios and edge cases are covered

If assiged this issue, I will ensure full compliance with the team and the team's guidelines.

## ETA - 36 hours

**Comment by hoangkianh** [comment]
I’d love to work on this task. The ETA is 1 hour

**Comment by morelucks** [comment]
Hi, I hope this message finds you well. My name is Kamshak Lucky Isuwa, a blockchain developer specializing in Solidity and Cairo smart contract development. I am passionate about contributing to impactful blockchain projects and am eager to assist in implementing a function mechanism to add supported ERC-20 tokens into the contract.
If granted this task, I aim to deliver a secure and efficient solution within 24 hours.
Best regards,
Kamshak Lucky Isuwa

**Comment by Jagadeeshftw** [comment]
Can I take this from here?

**Comment by FrankiePower** [comment]
I am Super Franky, a software developer with 3+ years of experience, and I have worked with frontend technologies like React, NextJS, JavaScript/TypeScript. I'm a recent graduate of Web3bridge Africa, the foremost and premier blockchain developer community.I am very familiar with smart contracts, and have built dapps. I have made contributions to many open-source projects, and I look forward to contributing to this one. My approach to implement a function mechanism to add supported ERC-20 tokens into the contract will be:

1. Review the current contract structure to understand where and how the tokens should be added.
2. Develop a function that allows for adding supported ERC-20 tokens to the contract by specifying necessary parameters like token address and associated metadata.
3. Test the function to ensure proper integration with the existing contract logic.

ET: 2 days
